### PR TITLE
API-2318: R4 RelatedPerson model

### PR DESCRIPTION
**Pull request by jjabando1**

### CODE REVIEW

**Review by jhulbert-va — CHANGES_REQUESTED** [review]
Samples and Tests

File: r4/src/main/java/gov/va/api/health/r4/api/RelatedPersonApi.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.enums.ParameterIn;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+
+public interface RelatedPersonApi {
```
Comment (jhulbert-va) [inline r4/src/main/java/gov/va/api/health/r4/api/RelatedPersonApi.java:15]: Can we move this to a "future" directory? Right now, we just want to get the models ready to go. The swagger and API definitions are not necessary right now as we will not be building a controller until later.

File: r4/src/main/java/gov/va/api/health/r4/api/resources/RelatedPerson.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+@Schema(
+    description = "https://www.hl7.org/fhir/R4/relatedperson.html",
+    example = "${r4.relper:gov.va.api.health.r4.api.swaggerexamples.SwaggerRelPer#relper}")
+public class RelatedPerson implements Resource {
+
+  // Ancestor -- Resource
+  @Pattern(regexp = Fhir.ID)
+  String id;
+
+  @NotBlank String resourceType;
```
Comment (jhulbert-va) [inline r4/src/main/java/gov/va/api/health/r4/api/resources/RelatedPerson.java:53]: Can we move this to the top and have a default of "RelatedPerson" (check out Lomboks builder defaults)?

File: r4/src/main/java/gov/va/api/health/r4/api/resources/RelatedPerson.java [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+@Builder
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+@AllArgsConstructor
+@JsonAutoDetect(
+    fieldVisibility = JsonAutoDetect.Visibility.ANY,
+    isGetterVisibility = JsonAutoDetect.Visibility.NONE)
+@Schema(
+    description = "https://www.hl7.org/fhir/R4/relatedperson.html",
+    example = "${r4.relper:gov.va.api.health.r4.api.swaggerexamples.SwaggerRelPer#relper}")
+public class RelatedPerson implements Resource {
```
Comment (jhulbert-va) [inline r4/src/main/java/gov/va/api/health/r4/api/resources/RelatedPerson.java:47]: RelatedPerson actually implements DomainResource

File: r4/src/main/java/gov/va/api/health/r4/api/resources/RelatedPerson.java [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+  @NotNull @Valid Reference patient;
+  @Valid List<CodeableConcept> relationship;
+  @Valid List<HumanName> name;
+  @Valid List<ContactPoint> telecom;
+  @Valid Gender gender;
+
+  @Pattern(regexp = Fhir.DATE)
+  String birthDate;
+
+  @Valid List<Address> locationAddress;
```
Comment (jhulbert-va) [inline r4/src/main/java/gov/va/api/health/r4/api/resources/RelatedPerson.java:81]: This field should be `address` check out the JsonProperty annotation.

File: r4/src/main/java/gov/va/api/health/r4/api/resources/RelatedPerson.java [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+  @Valid List<Attachment> photo;
+  @Valid Period period;
+
```
Comment (jhulbert-va) [inline r4/src/main/java/gov/va/api/health/r4/api/resources/RelatedPerson.java:84]: `communication` field.

File: r4/src/main/java/gov/va/api/health/r4/api/resources/RelatedPerson.java [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+  @Data
+  @NoArgsConstructor
+  @EqualsAndHashCode(callSuper = true)
+  @JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+  @JsonDeserialize(builder = RelatedPerson.Bundle.BundleBuilder.class)
+  @Schema(
+      name = "RelatedPersonBundle",
+      example =
+          "${r4.relperBundle:gov.va.api.health.r4.api.swaggerexamples.SwaggerRelPer#rpBundle}")
+  public static class Bundle extends AbstractBundle<Coverage.Entry> {
```
Comment (jhulbert-va) [inline r4/src/main/java/gov/va/api/health/r4/api/resources/RelatedPerson.java:102]: Copy Pasta!

**Review by jhulbert-va — CHANGES_REQUESTED** [review]

File: r4/src/main/java/gov/va/api/health/r4/api/resources/RelatedPerson.java [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+  public static class Communication implements BackboneElement {
+
+    @Pattern(regexp = Fhir.ID)
+    String id;
+
+    @Valid List<Extension> extension;
+
+    @Valid List<Extension> modifierExtension;
+
+    @NotNull CodeableConcept language;
```
Comment (jblefkove-va) [inline r4/src/main/java/gov/va/api/health/r4/api/resources/RelatedPerson.java:150]: `@Valid`

**Review by ilaflamme-va — APPROVED** [review]
lgtm

**Review by jhulbert-va — APPROVED** [review]

**Comment by jhulbert-va** [comment]
:shipit:

**Pull request merged**